### PR TITLE
Overhaul of structure to use DAOs and database for backend storage

### PR DESCRIPTION
**Pull request by bookdude13**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -19,12 +19,16 @@ Command-line interface for Proton
 Usage:
   ./proton init <folder> <root-public-key>
```
Comment (computergeek125) [inline src/main.rs:20]: Is there such a thing as argv[0] to get the exact program name instead of hardcoding one?

Comment (bookdude13) [reply src/main.rs:20]: There is, but this usage string is statically created and then used by Docopt, and Docopt uses this format for their example.

File: src/project_types/sequence.rs [hunk 2]
```diff
@@ -1,67 +1,79 @@
...
 impl Sequence {
-    /// Creates a new Sequence, allowing default values
-    /// of 50ms for frame_duration_ms and 1 for num_sections
-    /// Also initializes section file(s)
-    pub fn new(name: &str,
-        seq_directory_name: &str,
-        music_file_name: &str,
+    /// Creates a new Sequence, allowing the default value of 50ms for frame_duration_ms
+    pub fn new(
+        uid: u32,
```
Comment (computergeek125) [inline src/project_types/sequence.rs:24]: sid?

Comment (computergeek125) [reply src/project_types/sequence.rs:24]: nvm didn't read far enough

**Review by computergeek125 — APPROVED** [review]
Nothing looks awful 😛

**Pull request merged**